web/template/add_payment.gohtml

```diff
@@ -0,0 +1,61 @@
+
+{{ define "title" }}{{ if .Error.Any }}Error: {{ end }}Add payment{{ end }}
+
+{{ define "main" }}
+    <div class="govuk-grid-row">
+        <div class="govuk-grid-column-two-thirds">
+            <p class="govuk-body"><strong>{{ if eq .Case.SubType "pfa" }}PFA{{ else if eq .Case.SubType "lpa" }}LPA{{ end }} {{ .Case.UID }}</strong></p>
+
+            {{ template "error-summary" .Error }}
+
+            {{ if .Success }}
+                {{ template "success-banner" ("Payment added!") }}
+            {{ end }}
+
+            <h1 class="govuk-heading-l app-!-embedded-hide">Add a payment</h1>
+
+            <form class="form" method="POST">
+                <input type="hidden" name="xsrfToken" value="{{ .XSRFToken }}"/>
+                <div class="govuk-form-group">
+                    <h3 class="govuk-label-wrapper">
+                        <label class="govuk-label" for="amount">
+                            Enter amount
+                        </label>
+                    </h3>
+                    {{ template "errors" .Error.Field.amount }}
+                    <div class="govuk-input__wrapper">
+                        <div class="govuk-input__prefix" aria-hidden="true">£</div>
+                        <input class="govuk-input govuk-input--width-5 {{ if .Error.Field.amount }}govuk-input--error{{ end }}" id="amount" name="amount" type="number" step="0.01" value="{{ if ne .Amount "0" }}{{ .Amount }}{{ end }}">
+                    </div>
+                </div>
+
+                <div class="govuk-form-group">
+                    <label class="govuk-label" for="source">
+                        Payment method
+                    </label>
+                    {{ template "errors" .Error.Field.source }}
+                    <select class="govuk-select {{ if .Error.Field.source }}govuk-select--error{{ end }}" id="source" name="source">
+                        <option hidden selected disabled></option>
+                        <option value="PHONE" {{ if eq .Source "PHONE" }}selected{{ end }}>Paid over the phone</option>
+                        <option value="ONLINE" {{ if eq .Source "ONLINE" }}selected{{ end }}>Paid online - Notify</option>
+                    </select>
+                </div>
+
+                <div class="govuk-form-group govuk-!-display-inline-block govuk-!-padding-right-4 {{ if .Error.Field.paymentDate }}govuk-form-group--error{{ end }}">
+                    <label class="govuk-label" for="f-paymentDate">Date of payment</label>
+                    {{ template "errors" .Error.Field.paymentDate }}
+                    <input class="govuk-input date-picker {{ if .Error.Field.paymentDate }}govuk-input--error{{ end }}" type="date" id="f-paymentDate" name="paymentDate" value="{{ .PaymentDate }}"/>
+                </div>
+
+                <a href="javascript:void(0)" class="govuk-link govuk-link--no-visited-state govuk-!-display-inline-block" data-module="select-todays-date">Use today's date</a>
```

Comment:
Prefer `#` as inline JavaScript can often cause security warnings in scanning tools and occasionally in browsers. This requires the "preventDefault" I mentioned above so that it doesn't actually try to link to `/lpa/frontend/add-payment?id=1234#`
```suggestion
                <a href="#" class="govuk-link govuk-link--no-visited-state govuk-!-display-inline-block" data-module="select-todays-date">Use today's date</a>
```

Reply:
Ahh I didn't consider the security warnings on inline JS, that's a good point, thanks! :)